web/pandas/pdeps/0017-backwards-compatibility-and-deprecation-policy.md

```diff
@@ -0,0 +1,84 @@
+# PDEP-17: Backwards compatibility and deprecation policy
+
+- Created: 27 June 2024
+- Status: Under discussion
+- Discussion: [#59125](https://github.com/pandas-dev/pandas/issues/59125)
+- Author: [Abdulaziz Aloqeely](https://github.com/Aloqeely)
+- Revision: 1
+
+## Abstract
+
+This PDEP defines pandas' backwards compatibility and deprecation policy.
+
+## Motivation
+
+Having a clear backwards compatibility and deprecation policy is crucial to having a healthy ecosystem. We want to ensure users can rely on pandas being stable while still allowing the library to evolve.
+
+This policy will ensure that users have enough time to deal with deprecations while also minimizing disruptions on downstream packages' users.
+
+## Scope
+
+This PDEP covers pandas' approach to backwards compatibility and the deprecation and removal process.
+
+The decision making process for deprecating features is out of scope for this PDEP.
+
+## Background
+
+pandas uses a loose variant of semantic versioning.  
+A pandas release number is written in the format of ``MAJOR.MINOR.PATCH``.
+
+For the purposes of this PDEP, the last minor release before a major release will be referred to as the 'final minor version'.
+
+## General policy
+
+This policy applies to the public API. Anything not part of the public API or is marked as "Experimental" may be changed or removed at anytime.
+
+- Breaking backwards compatibility should benefit more than it harms users.
+- Breaking changes should go through a deprecation cycle before being implemented.
+- Breaking changes should only occur in major releases.
+- No deprecations should be introduced in patch releases.
```

Comment:
What do you think the benefit of this is? My initial feeling is that this seems too strict, but curious what the motivation for this is

Reply:
Which point are you referring to exactly? If it's the last one, that seems to be the current approach pandas is following (from the [version policy](https://pandas.pydata.org/docs/development/policies.html#version-policy) page)

Also, that follows rule 7 of the SemVer specification. 

> Minor version Y (x.Y.z | x > 0) MUST be incremented if new, backward compatible functionality is introduced to the public API. **It MUST be incremented if any public API functionality is marked as deprecated.**